lib/main.coffee

```diff
@@ -31,9 +26,20 @@ module.exports =
     @subscriptions.add atom.config.observe 'linter-phpmd.rulesets',
       (rulesets) =>
         @rulesets = rulesets
-    @subscriptions.add atom.config.observe 'linter-phpmd.projectRules',
-      (projectRules) =>
-        @projectRules = projectRules
+    if atom.config.get('linter-phpmd.projectRules') == true &&
+    atom.config.get('linter-phpmd.rulesets') ==
+    atom.config.defaultSettings['linter-phpmd'].rulesets
+      atom.config.unset('linter-phpmd.projectRules')
+      atom.config.set('linter-phpmd.rulesets', 'ruleset.xml')
+    else if atom.config.get('linter-phpmd.projectRules') == true
```

Comment:
You're checking this condition twice, since it is unset in either case just put this whole block inside an `if` block of this conditional.